### DISCUSSION TITLE
Lab 9 Checkpoint 0

### DISCUSSION THREAD

**Discussion by roryeiffe** [roryeiffe/Fresh-Results#21] [General]
MongoDB's license change is devastating to developers who want to use MongoDB in their own project. The license change seems innocent enough, especially according to how the MongoDB website words it. They ensure that the SSPL license still grants users the freedom to "use, review, modify, and redistribute the source code." The only difference would be changing the terms for offering MongoDB as a service. However, according to Peter Zaitsev, the SSPL license is harmful to not only cloud vendors, but all MongoDB users. Zaitsev says that under the SSPL, people who would want to offer MongoDB as a service would either have to release all related software or purchase a commercial license from Mongo. For cloud vendors, releasing all related software is impractical and the need to purchase a commercial license goes against the spirit of open-source in my opinion, especially if the price goes up in the coming years. As it is, the highest tier of MongoDB Atlas costs $95/month. Of course, it is understandable that Mongo wants to make more money from their software, especially since so many people use it. It is also important to remember that they had every legal right to do so. However, the SSPL might have opened a door that can't be closed, and could cause MongoDB to be difficult to incorporate into future projects, most prominently with development teams who don't have a lot of money. The controversy surrounding this license change just shows that money makes open-source-software more complicated. If money wasn't an issue, Mongo would probably have no issue keeping their license as nonrestrictive as possible.

Works Cited:

Peter.zaitsev. “Why Is MongoDB's SSPL Bad For You?” Percona Database Performance Blog, 5 Feb. 2021, www.percona.com/blog/2020/06/16/why-is-mongodbs-sspl-bad-for-you/. 

“Server Side Public License FAQ.” MongoDB, www.mongodb.com/licensing/server-side-public-license/faq.

**Comment by SerenaChen1** [comment]
It's understandable that MongoDB decided to change its license to the SSPL, as many cloud providers have been hosting commercial versions of its database without contributing anything to the open source community. For the most part, regular users are unaffected, as this license applies mainly to cloud platform providers.  However, there seems to be a shared concern among MongoDB's end users that the license is too general in what is considered a derivation of MongoDB, as it could include monitoring, backup, analytics, etc. So while MongoDB argues that regular developers will not be affected, they could easily make a claim against a user's use of the software. Many also contend that the immediate switch to the SSPL was not respectful to users, especially because the license is not an OSI approved license. As a result, many users had to go through a review process to ensure the license compatibility, which took up several weeks. During this time, users weren't able to access to the software. If MongoDB truly cared for their users, they would've announced the license change earlier so their users would have time to adjust, so despite the fact that MongoDB claims SSPL won't affect regular users, people must be wary of what their intentions are.

Works Cited

“MongoDB's Server Side Public License Is Likely Unenforceable: Hacker News.” MongoDB's Server Side Public License Is Likely Unenforceable | Hacker News, https://news.ycombinator.com/item?id=18301116

Zaitsev, Peter. “Percona Statement on MongoDB Community Server License Change.” Percona Database Performance Blog, 7 Aug. 2019, www.percona.com/blog/2018/10/18/percona-statement-on-mongodb-community-server-license-change/.

**Comment by VictoriaSavage526** [comment]
The Supreme Court recently ruled that Google's usage of Oracle's API constituted Fair Use. The ruling was 6-2, with Justices Sotomayor, Kagan, Kavanagh, Breyer, Gorsuch, and Chief Justice Roberts in the majority, and Justices Thomas and Alito dissenting. The Open Source Initative filed an Amicus Curiae brief in support of Google. If this case was decided a different way, it is likely that Open Source Software as we know it would be jeopardized. Justice Stephen Breyer wrote that due to the cross-functionality of utilizing the code, and only using 2.3% of the API, Google meaningfully transformed the code. It is likely that licenses that do not wish to allow a la carte copying and repurposing will have to be transformed in accordance with precedent. Additionally, interoperability, had the opinion of Alito and Thomas been the majority, would have been at risk. If an API can't be used, interusage and interoperability would be in jeopardy and would damage the open source community.

Works Cited

Google vs Oracle: Resolved in Favor of Open Source, The Open Source Initiative, https://opensource.org/googlevoracleresolved 

The Supreme Court Issued Google v. Oracle Copyright Decision, The National Law Review. https://www.natlawreview.com/article/supreme-court-issued-google-v-oracle-copyright-decision
